jax/interpreters/xla.py

```diff
@@ -200,7 +198,7 @@ class and an int id (representing a Device instance) or a None.
     raise ValueError(msg.format(", ".join(names)))
   else:
     all_devices = it.chain(xb.devices(), xb.devices('cpu'))
-    return device and next(d for d in all_devices if (type(d), d.id) == device)
+    return device and next(d for d in all_devices if d == device)
```

Comment:
I haven't paged it back in fully, but I think this function should be simplified more. This weird logic (to return an element of `all_devices` equal to `device`, rather than just `device`) existed only because of the tuple stuff.